share/eds/byzantine/bad_encoding_test.go

```diff
@@ -37,10 +38,92 @@ func TestBadEncodingFraudProof(t *testing.T) {
 	errByz := NewErrByzantine(ctx, bServ, &dah, errRsmt2d)
 
 	befp := CreateBadEncodingProof([]byte("hash"), 0, errByz)
-	err = befp.Validate(&header.ExtendedHeader{
-		DAH: &dah,
-	})
-	assert.NoError(t, err)
+
+	var test = []struct {
+		name           string
+		doFn           func() error
+		expectedResult func(error)
+	}{
+		{
+			name: "valid BEFP",
+			doFn: func() error {
+				return befp.Validate(&header.ExtendedHeader{DAH: &dah})
+			},
+			expectedResult: func(err error) {
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "heights mismatch",
+			doFn: func() error {
+				return befp.Validate(&header.ExtendedHeader{
+					RawHeader: core.Header{
+						Height: 42,
+					},
+					DAH: &dah,
+				})
+			},
+			expectedResult: func(err error) {
+				require.Error(t, err)
+			},
+		},
+		{
+			name: "index out of bounds",
+			doFn: func() error {
+				befp, ok := befp.(*BadEncodingProof)
+				require.True(t, ok)
+				befpCopy := *befp
+				befpCopy.Index = 100
+				return befpCopy.Validate(&header.ExtendedHeader{DAH: &dah})
+			},
+			expectedResult: func(err error) {
+				require.Error(t, err)
+			},
```

Comment:
Do you want to assert certain things about the error too?

Reply:
I don't think so. We don't have specific errors in `Validate`

Reply:
[blocking]
We should introduce them then. When I was writing unit tests in the go-header for Verify, I first similarly omitted custom error sentinels but then saw that some tests returned errors for reasons different from the setup after code changes, which made me introduce errors. This errors do not have to be public, but they are necessary to ensure units are correct.